### PR TITLE
Small, stylistic changes

### PR DESCRIPTION
**Pull request by stephencelis**
Happy to drop any of the commits or the PR entirely. Was just reading the code and found a few spots that could become shorter and less mutable without hurting readability.

### CODE REVIEW

File: DVR/Cassette.swift [hunk 1]
```diff
@@ -27,7 +27,7 @@ extension Cassette {
     var dictionary: [String: AnyObject] {
         return [
             "name": name,
-            "interactions": interactions.map() { $0.dictionary }
+            "interactions": interactions.map { $0.dictionary }
```
Comment (soffes) [inline DVR/Cassette.swift:30]: Oh who knew?! Is this new?

File: DVR/Cassette.swift [hunk 2]
```diff
@@ -36,14 +36,10 @@ extension Cassette {
...
         self.name = name
 
-        var interactions = [Interaction]()
         if let array = dictionary["interactions"] as? [[String: AnyObject]] {
-            for dictionary in array {
-                if let interaction = Interaction(dictionary: dictionary) {
-                    interactions.append(interaction)
-                }
-            }
+            interactions = array.flatMap { Interaction(dictionary: $0) }
```
Comment (soffes) [inline DVR/Cassette.swift:40]: What?! Heck yes.

Comment (soffes) [inline DVR/Cassette.swift:40]: I'm going to use this everywhere. Amazing.

File: DVR/URLResponse.swift [hunk 3]
```diff
@@ -17,13 +17,11 @@ class URLResonse: NSURLResponse {
...
     var dictionary: [String: AnyObject] {
-        var dictionary = [String: AnyObject]()
-
         if let url = URL?.absoluteString {
-            dictionary["url"] = url
+            return ["url": url]
         }
 
-        return dictionary
+        return [:]
```
Comment (soffes) [inline DVR/URLResponse.swift:24]: That's a little weird, but I'm into it.

**Comment by soffes** [comment]
:heart: :heart: :heart: :heart: :heart: :heart: :heart:

File: DVR/Cassette.swift (hunk 2 above)
Comment (remirobert) [inline DVR/Cassette.swift:40]: this is very great !

File: DVR/Cassette.swift (hunk 1 above)
Comment (subdigital) [inline DVR/Cassette.swift:30]: I've been using this since day 1.  I love trailing closure syntax.  :tada:

File: DVR/URLResponse.swift (hunk 3 above)
Comment (jbrooksuk) [inline DVR/URLResponse.swift:24]: :+1: I love how you love Swifts quirky features.

Comment (natebird) [inline DVR/URLResponse.swift:24]: Just to clarify, what exactly is this, `return [:]`, returning?

File: DVR/Cassette.swift (hunk 1 above)
Comment (soffes) [inline DVR/Cassette.swift:30]: Didn't realize the `()` were optional if you did that. So great.

File: DVR/URLResponse.swift (hunk 3 above)
Comment (jbrooksuk) [inline DVR/URLResponse.swift:24]: Assuming it's `[String: AnyObject]`

Comment (natebird) [inline DVR/URLResponse.swift:24]: Ok. Great! Thanks for clarifying.

Comment (stephencelis) [inline DVR/URLResponse.swift:24]: @natebird `[:]` is just an empty dictionary literal. The function signature allows the underlying type to be inferred.

**Pull request merged**